packages/flutter/lib/src/widgets/form.dart

```diff
@@ -291,18 +291,46 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  ///
+  /// See also:
+  ///  * [validateGranularly], which also validates descendant [FormField]s,
+  /// but instead returns a [Set] of fields with errors
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
     return _validate();
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField] that is a descendant of this [Form], and
+  /// returns a [Set] of [FormFieldState] of the invalid field(s) only, if any.
+  ///
+  /// This method can be useful to highlight field(s) with errors,
+  /// possibly using their widget key(s).
+  ///
+  /// The form will rebuild to report the results.
+  ///
+  /// See also:
+  ///  * [validate], which also validates descendant [FormField]s,
+  /// and return true if there are no errors
```

Comment:
missing "." at the end.

Reply:
nit: "but returns a boolean value indicating whether the [Form] has invalid user inputs" or something like that?

Reply:
That's actually the docs for `validate`, it was there before this pr, I just copied it to the "see also".